update contributing guides

jukent opened the pull request:
<!--
Project Pythia has automated review requests. If you are not ready for reviews on this contribution, please
open this Pull Request as a "Draft" by clicking the dropdown on the green "Create Pull Request" button in the
lower right corner here.
-->

jukent commented:
> Related: #428, #431, #447, #518, #534

Thanks for reminding me of these issues @r-ford I'm off for the day but will look at making everything consistent tomorrow.

brian-rose commented:
I have some reservations about renaming several of these files because that will likely break existing links. Since we're not running full link checking right now, we may not spot all broken internal links, and may also inadvertently break external links to our content. So I think there needs to be a cost-benefit consideration.

jukent commented:
Heard @brian-rose I changed the file name back.

I updated the language and a few other things.

Any insight why it looks like I'm adding a `myst.yml` file? I don't remember us removing one or why we wouldn't have needed it.

erogluorhan reviewed (approved):
Looks good to me; thanks for this!

erogluorhan commented:
> Any insight why it looks like I'm adding a `myst.yml` file? I don't remember us removing one or why we wouldn't have needed it.

Actually, from the history, `myst.yml` seems to have been moved into `/portal` with [PR#519](https://github.com/ProjectPythia/projectpythia.github.io/pull/519). 

With this PR got merged, we have a duplicate file now in the root as well.

brian-rose commented:
#519 was necessary to restore the correct URLs for pages on the portal site. Also keeping `myst.yml` inside the `/portal` directory and running the myst build from there keeps the build process consistent with our previous portal, which means fewer edits to contributors guides etc.
